### PR TITLE
Add experimental audit for high network idle times in ad loading

### PR DESCRIPTION
**Pull request by warrengm**
High network idle times in the ad loading graph can indicate many bad practices (timeouts, waiting on load events, long tasks, etc.) so this audit checks for any egregious cases.

I'm labeling this audit as experimental since it's not clear what the false positive/negative rates are. Additionally this audit lacks attribution currently, but users can cross reference this audit with the long tasks audit, etc. I will look into including possible culprits as a follow-up

### CODE REVIEW

File: lighthouse-plugin-ad-speed-insights/audits/idle-network-times.js [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+    const criticalRequests = getCriticalPath(
+      networkRecords, adRequest, trace.traceEvents);
+
+    const pageStartTime = getPageStartTime(networkRecords);
+    const blockingRequests = Array.from(criticalRequests)
+        .filter((r) => ['Script', 'XHR', 'Fetch', 'EventStream', 'Document'].includes(r.resourceType))
+        .filter((r) => r.mimeType != 'text/css')
+        .filter((r) => r.startTime > 0)
+        .map((r) => ({
+          startTime: (r.startTime - pageStartTime) * 1e3,
```
Comment (jburger424) [inline lighthouse-plugin-ad-speed-insights/audits/idle-network-times.js]: Nit: In my opinion 1e3 is less readable than 1000 but it matters very little.

File: lighthouse-plugin-ad-speed-insights/audits/idle-network-times.js [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+
+    if (!blockingRequests) {
+      return auditNotApplicable('No ads requested');
+    }
+
+    let maxEndSoFar = Infinity;
+    const idleTimes = [];
+    for (let i = 0; i < blockingRequests.length;) {
+      const {startTime, endTime} = blockingRequests[i];
+      if (startTime - maxEndSoFar > 150) { // Tune this!
```
Comment (jburger424) [inline lighthouse-plugin-ad-speed-insights/audits/idle-network-times.js]: // Tune this! => // TODO...
Same below

File: lighthouse-plugin-ad-speed-insights/plugin.js [hunk 3]
```diff
@@ -54,6 +55,7 @@ module.exports = {
       {id: 'ad-blocking-tasks', weight: 1, group: 'ads-performance'},
       {id: 'ad-request-critical-path', weight: 1, group: 'ads-performance'},
       {id: 'serial-header-bidding', weight: 1, group: 'ads-performance'},
+      {id: 'idle-network-times', weight: 0.5, group: 'ads-performance'},
```
Comment (jburger424) [inline lighthouse-plugin-ad-speed-insights/plugin.js]: What are your thoughts on splitting this into a new experimental group? The group could be displayed based on a `--experimental` flag.

Comment (warrengm) [reply lighthouse-plugin-ad-speed-insights/plugin.js]: SGTM, but I'll defer it for a future PR. It's also worth having a broader discussion around weighing audits, etc.

File: lighthouse-plugin-ad-speed-insights/audits/idle-network-times.js (hunk 1 above)
Comment (warrengm) [reply lighthouse-plugin-ad-speed-insights/audits/idle-network-times.js]: It gets a lot more readable when multiplying. E.g. I know nanoseconds are milliseconds  * 10^-6.

File: lighthouse-plugin-ad-speed-insights/plugin.js (hunk 3 above)
Comment (jburger424) [reply lighthouse-plugin-ad-speed-insights/plugin.js]: Could you put up an issue or add a TODO?

**Pull request merged**